### ISSUE TITLE
[packagers] Broken java-archive launcher in windows

### ISSUE THREAD

**Issue by jvalkeal** [jreleaser/jreleaser#1634]
Originally in https://github.com/spring-projects/spring-cli/issues/178 but noticed same issue in jreleaser as well. `spring-cli` is using fatjar while jreleaser uses exploded jar but both ways fail with:

```
Error: Could not find or load main class
Caused by: java.lang.ClassNotFoundException:
```
Problem is in https://github.com/jreleaser/jreleaser/blob/5ba315d18b85e3937c43663fc418e03cf4b13899/core/jreleaser-templates/src/main/resources/META-INF/jreleaser/templates/java-archive/java-archive/bin/launcher.bat.tpl#L57 which is causing extra argument with space.

I took [jreleaser-standalone-early-access-windows-x86_64.zip](https://github.com/jreleaser/jreleaser/releases/download/early-access/jreleaser-standalone-early-access-windows-x86_64.zip) and enabled "echo" in bat file:

```
C:\Users\jannev\Downloads\jreleaser-standalone-early-access-windows-x86_64\jreleaser-standalone-early-access-windows-x86_64\bin>set JAVA_OPTS=" "

C:\Users\jannev\Downloads\jreleaser-standalone-early-access-windows-x86_64\jreleaser-standalone-early-access-windows-x86_64\bin>""C:\Users\jannev\Downloads\jreleaser-standalone-early-access-windows-x86_64\jreleaser-standalone-early-access-windows-x86_64"\bin\java"  " " -cp """C:\Users\jannev\Downloads\jreleaser-standalone-early-access-windows-x86_64\jreleaser-standalone-early-access-windows-x86_64\jars"\*"" org.jreleaser.cli.Main
Error: Could not find or load main class
Caused by: java.lang.ClassNotFoundException:
```

In `spring-cli` it looks like:

```
C:\Users\jannev\Downloads\spring-cli-standalone-0.9.0-SNAPSHOT-windows-x86_64>set JAVA_OPTS=" "

C:\Users\jannev\Downloads\spring-cli-standalone-0.9.0-SNAPSHOT-windows-x86_64>""C:\Users\jannev\Downloads\spring-cli-standalone-0.9.0-SNAPSHOT-windows-x86_64"\bin\java"  " " -cp ""C:\Users\jannev\Downloads\spring-cli-standalone-0.9.0-SNAPSHOT-windows-x86_64\jars"\*" -jar ""C:\Users\jannev\Downloads\spring-cli-standalone-0.9.0-SNAPSHOT-windows-x86_64\jars"\spring-cli-0.9.0-SNAPSHOT.jar"
Error: Could not find or load main class
Caused by: java.lang.ClassNotFoundException:
```

You get same error in java by just doing:
```
$ java " "
Error: Could not find or load main class  
Caused by: java.lang.ClassNotFoundException:  
```

I could try to PR a fix?

**Comment by aalmiray** [comment]
We'd very much welcome a PR, thank you 😅

**Comment by aalmiray** [comment]
FWIW I think this could be fixed by using mustache itself, like so

```
{{#distributionJavaOptions}}
 set JAVA_OPTS="%JAVA_OPTS% {{.}}" 
{{/distributionJavaOptions}}
```

**Comment by jvalkeal** [comment]
That was something what I was thinking as well. However if you read that scala issue quoting is problematic itself if it'd done blindly as it breaks if existing JAVA_OPTS itself contain quotes. That's what I was planning to experiment that when this expansion actually blows up.

**Comment by aalmiray** [comment]
AFAICT package processors should ensure the value of JAVA_OPTS is properly quoted before sending it to the Mustache template engine.

**Comment by jvalkeal** [comment]
Actually there's additional issues for quoting `-cp` value. Looking older jreleaser versions, `1.9.0` is a last good release. `1.10.0` works if I remove `JAVA_OPTS` from `jreleaser.bat`. `1.11.0` fails with wrong quotes even if I remove `JAVA_OPTS`.

This was in my origin comment but I didn't pay attention(three quotes where one after jars messes things up):

```
-cp """C:\Users\jannev\Downloads\jreleaser-standalone-early-access-windows-x86_64\jreleaser-standalone-early-access-windows-x86_64\jars"\*""
```

I didn't realise this with `spring-cli` as while while `-cp` is wrongly quoted it doesn't matter as it's using fatjar.

**Comment by jvalkeal** [comment]
For JAVA_OPTS I think we need to remove quoting as it will always break things up if there's more than one option(or one option with extra space).

```
{{#distributionJavaOptions}}
 set JAVA_OPTS=%JAVA_OPTS% {{.}}
{{/distributionJavaOptions}}
```
If we want to be pedantic:
```
{{#distributionJavaOptions}}
if not defined JAVA_OPTS (
  set JAVA_OPTS={{.}}
) else (
  set JAVA_OPTS=%JAVA_OPTS% {{.}}
)
{{/distributionJavaOptions}}
```

As bat interpreter doesn't remove quotes when using `set` these would fail:
```
java "-Xms1g -Xmx1G"
java "-Xms1g "
```
you can use quotes but those needs to immediately surround options like:

```
java "-Xms1g" "-Xmx1G"
```

And this same story goes with `-cp`.

**Comment by aalmiray** [comment]
I see. Individual elements may need to be quoted but not the full set as a whole (all entries seen as a single argument).

**Comment by aalmiray** [comment]
🎉 This issue has been resolved in `v1.12.0` ([Release Notes](https://github.com/jreleaser/jreleaser/releases/tag/v1.12.0))
